src/pini.cpp

```diff
@@ -89,4 +89,11 @@ void pini::default_callback(std::string_view msg, severity level) {
 	out << msg << '\n';
 }
 
+bool pini::is_empty() { return key_value_pairs.empty(); }
+std::size_t pini::size() { return key_value_pairs.size(); }
```

Comment:
You may want to inline `is_empty()` and `size()` (implement in header / in class definition)